dhis-2/dhis-services/dhis-service-core/src/test/java/org/hisp/dhis/fileresource/FileResourceBlacklistTest.java

```diff
@@ -0,0 +1,64 @@
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.hisp.dhis.fileresource;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+
+import org.junit.Test;
+
+/**
+ * @author Lars Helge Overland
+ */
+public class FileResourceBlacklistTest
+{
+    @Test
+    public void testValid()
+    {
+        FileResource frA = new FileResource( "My_Checklist.pdf", "application/pdf", 324, "", FileResourceDomain.DATA_VALUE );
+        FileResource frB = new FileResource( "Evaluation.docx", "application/vnd.openxmlformats-officedocument.wordprocessingml.document", 541, "", FileResourceDomain.MESSAGE_ATTACHMENT );
+        FileResource frC = new FileResource( "FinancialReport.xlsx", "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet", 143, "", FileResourceDomain.DATA_VALUE );
+
+        assertTrue( FileResourceBlacklist.isValid( frA ) );
+        assertTrue( FileResourceBlacklist.isValid( frB ) );
+        assertTrue( FileResourceBlacklist.isValid( frC ) );
+    }
+
+    @Test
+    public void testInvalid()
+    {
+        FileResource frA = new FileResource( "Click_Me.exe", "application/x-ms-dos-executable", 451, "", FileResourceDomain.DATA_VALUE );
+        FileResource frB = new FileResource( "evil_script.sh", "application/pdf", 125, "", FileResourceDomain.MESSAGE_ATTACHMENT ); // Fake content type
+        FileResource frC = new FileResource( "cookie_stealer", "text/javascript", 631, "", FileResourceDomain.USER_AVATAR ); // No file extension
+
+        assertFalse( FileResourceBlacklist.isValid( frA ) );
+        assertFalse( FileResourceBlacklist.isValid( frB ) );
+        assertFalse( FileResourceBlacklist.isValid( frC ) );
+    }
+}
```

Comment:
Should we add a test to cover the cases where the client passes in a null argument to the "isValid" method?

Reply:
Yes agree, will fix.